src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperAttribute.cs

```diff
@@ -8,77 +8,70 @@ namespace Microsoft.AspNetCore.Razor.TagHelpers
     /// <summary>
     /// An HTML tag helper attribute.
     /// </summary>
-    public class TagHelperAttribute : IReadOnlyTagHelperAttribute
+    public class TagHelperAttribute
     {
         private static readonly int TypeHashCode = typeof(TagHelperAttribute).GetHashCode();
 
         /// <summary>
-        /// Instantiates a new instance of <see cref="TagHelperAttribute"/>.
+        /// Instantiates a new instance of <see cref="TagHelperAttribute"/> with the specified <paramref name="name"/>.
+        /// <see cref="Minimized"/> is set to <c>true</c> and <see cref="Value"/> to <c>null</c>.
         /// </summary>
-        public TagHelperAttribute()
+        /// <param name="name">The <see cref="Name"/> of the attribute.</param>
+        public TagHelperAttribute(string name)
+            : this(name, value: null, minimized: true)
         {
         }
 
         /// <summary>
-        /// Instantiates a new instance of <see cref="TagHelperAttribute" /> with values provided by the given
-        /// <paramref name="attribute"/>.
+        /// Instantiates a new instance of <see cref="TagHelperAttribute"/> with the specified <paramref name="name"/>
+        /// and <paramref name="value"/>. <see cref="Minimized"/> is set to <c>false</c>.
         /// </summary>
-        /// <param name="attribute">A <see cref="IReadOnlyTagHelperAttribute"/> whose values should be copied.</param>
-        public TagHelperAttribute(IReadOnlyTagHelperAttribute attribute)
-            : this(attribute?.Name, attribute?.Value)
+        /// <param name="name">The <see cref="Name"/> of the attribute.</param>
+        /// <param name="value">The <see cref="Value"/> of the attribute.</param>
+        public TagHelperAttribute(string name, object value)
+            : this(name, value, minimized: false)
         {
-            if (attribute == null)
-            {
-                throw new ArgumentNullException(nameof(attribute));
-            }
-
-            Minimized = attribute.Minimized;
         }
 
         /// <summary>
-        /// Instantiates a new instance of <see cref="TagHelperAttribute"/> with the specified <paramref name="name"/>
-        /// and <paramref name="value"/>.
+        /// Instantiates a new instance of <see cref="TagHelperAttribute"/> with the specified <paramref name="name"/>,
+        /// <paramref name="value"/> and <paramref name="minimized"/>.
         /// </summary>
         /// <param name="name">The <see cref="Name"/> of the attribute.</param>
         /// <param name="value">The <see cref="Value"/> of the attribute.</param>
-        public TagHelperAttribute(string name, object value)
+        /// <param name="minimized">The <see cref="Minimized"/> value of the attribute.</param>
```

Comment:
Add something stating that `value` is completely ignored when `minimized: true`.